sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java

```diff
@@ -44,110 +43,87 @@
   }
 
   @Test
   public void noWarningsWhenGettingProtobufPathAndNoPropertyAvailable_TestProject() {
     // Test projects have sonar.tests property set, main projects don't
-    settings.setProperty("sonar.tests", "");
-    config = new AbstractProjectConfiguration(settings.asConfig(), "cs") {
-    };
+    Configuration configuration = createEmptyConfiguration();
+    when(configuration.hasKey("sonar.tests")).thenReturn(true);
+
+    AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).isEmpty();
     assertThat(logTester.logs(LoggerLevel.WARN)).isEmpty();
   }
 
   @Test
   public void giveWarningsWhenGettingProtobufPathAndNoFolderAvailable() throws IOException {
     Path path1 = createProtobufOut("report");
-    settings.setProperty("sonar.cs.analyzer.projectOutPaths", new String[] {path1.toString(), "non-existing"});
-    config = new AbstractProjectConfiguration(settings.asConfig(), "cs") {
-    };
+
+    Configuration configuration = createEmptyConfiguration();
+    when(configuration.getStringArray("sonar.cs.analyzer.projectOutPaths")).thenReturn(new String[] {path1.toString(), "non-existing"});
+
+    AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).containsOnly(path1.resolve("output-cs"));
     assertThat(logTester.logs(LoggerLevel.WARN)).hasSize(1);
     assertThat(logTester.logs(LoggerLevel.WARN).get(0)).matches(s -> s.startsWith("Analyzer working directory does not exist"));
   }
 
   @Test
   public void giveWarningsWhenGettingOldProtobufPathAndNoFolderAvailable() {
-    settings.setProperty("sonar.cs.analyzer.projectOutPath", "non-existing");
-    config = new AbstractProjectConfiguration(settings.asConfig(), "cs") {
-    };
+    Configuration configuration = createEmptyConfiguration();
+    when(configuration.get("sonar.cs.analyzer.projectOutPath")).thenReturn(Optional.of("non-existing"));
+
+    AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).isEmpty();
     assertThat(logTester.logs(LoggerLevel.WARN)).hasSize(1);
     assertThat(logTester.logs(LoggerLevel.WARN).get(0)).matches(s -> s.startsWith("Analyzer working directory does not exist"));
   }
 
   @Test
   public void informHowManyProtoFilesAreFound() throws IOException {
-    setProtobufOut();
-    config = new AbstractProjectConfiguration(settings.asConfig(), "cs") {
-    };
+    Configuration configuration = createEmptyConfiguration();
+    mockProtobufOutPaths(configuration);
+
+    AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).isNotEmpty();
     assertThat(logTester.logs(LoggerLevel.DEBUG)).hasSize(2);
```

Comment:
should we also check the `DEBUG` messages? 
also, for the WARN below, should it rather be `hasOnlyOneElementSatisfying` or does it have more messages? Do we know how many?

Reply:
this test confuses me together with `onlyProtobufReportsPresent` - the mock has 2 files only - why is it logging that it doesn't find something?

I'm trying to understand the tests in order to see if something is missing

Reply:
This test is misleading, `logTester.logs(LoggerLevel.WARN)` is actually empty (there is no warning). I will change the assert to clarify that.

Reply:
> Regarding onlyProtobufReportsPresent, I'm not sure what you mean. There are no logs verified in it.

it does the exact same actions as `informHowManyProtoFilesAreFound` . I meant the 2 tests that are testing the same behavior (in different ways) are confusing.